tests/basic_one_way_send.go

```diff
@@ -0,0 +1,156 @@
+	Expect(err).Should(BeNil())
+
+	log.Info("Sending Teleporter transaction on source chain", "destinationChainID", subnetBInfo.BlockchainID, "txHash", signedTx.Hash())
+	newHeadA := SendAndWaitForTransaction(ctx, subnetAInfo.ChainWSClient, signedTx)
+
+	receipt, err := subnetAInfo.ChainWSClient.TransactionReceipt(ctx, signedTx.Hash())
+	Expect(err).Should(BeNil())
+	Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+
+	//
+	// Relay the message to the destination
+	//
+
+	// Get the latest block from Subnet A, and retrieve the warp message from the logs
+	log.Info("Waiting for new block confirmation")
+	blockHashA := newHeadA.Hash()
+
+	log.Info("Fetching relevant warp logs from the newly produced block")
+	logs, err := subnetAInfo.ChainWSClient.FilterLogs(ctx, interfaces.FilterQuery{
+		BlockHash: &blockHashA,
+		Addresses: []common.Address{warp.Module.Address},
+	})
+	Expect(err).Should(BeNil())
+	Expect(len(logs)).Should(Equal(1))
+
+	// Check for relevant warp log from subscription and ensure that it matches
+	// the log extracted from the last block.
+	txLog := logs[0]
+	log.Info("Parsing logData as unsigned warp message")
+	unsignedMsg, err := avalancheWarp.ParseUnsignedMessage(txLog.Data)
+	Expect(err).Should(BeNil())
+
+	// Set local variables for the duration of the test
+	unsignedWarpMessageID := unsignedMsg.ID()
+	unsignedWarpMsg := unsignedMsg
+	log.Info("Parsed unsignedWarpMsg", "unsignedWarpMessageID", unsignedWarpMessageID, "unsignedWarpMessage", unsignedWarpMsg)
+
+	// Loop over each client on chain A to ensure they all have time to accept the block.
+	// Note: if we did not confirm this here, the next stage could be racy since it assumes every node
+	// has accepted the block.
+	WaitForAllValidatorsToAcceptBlock(ctx, subnetAInfo.ChainNodeURIs, subnetAInfo.BlockchainID, newHeadA.Number.Uint64())
+
+	// Get the aggregate signature for the Warp message
+	log.Info("Fetching aggregate signature from the source chain validators")
+	warpClient, err := warpBackend.NewClient(subnetAInfo.ChainNodeURIs[0], subnetAInfo.BlockchainID.String())
+	Expect(err).Should(BeNil())
+	signedWarpMessageBytes, err := warpClient.GetAggregateSignature(ctx, unsignedWarpMessageID, params.WarpQuorumDenominator)
+	Expect(err).Should(BeNil())
+
+	// Construct the transaction to send the Warp message to the destination chain
+	signedTxB := ConstructAndSendTransaction(
+		ctx,
+		signedWarpMessageBytes,
+		big.NewInt(1),
+		teleporterContractAddress,
+		fundedAddress,
+		fundedKey,
+		subnetBInfo.ChainWSClient,
+		subnetBInfo.ChainIDInt,
+	)
+
+	receipt, err = subnetBInfo.ChainWSClient.TransactionReceipt(ctx, signedTxB.Hash())
+	Expect(err).Should(BeNil())
+	Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+
+	sendCrossChainMessageLog := receipt.Logs[0]
+	var event SendCrossChainMessageEvent
+	err = teleporter.EVMTeleporterContractABI.UnpackIntoInterface(&event, "SendCrossChainMessage", sendCrossChainMessageLog.Data)
```

Comment:
why does this unpack into sendCrossChainmessage, shouldn't it be `receiveCrossChainMessage`

Reply:
This is the `SendCrossChainMessage` event on the sending subnet. We use it to get the Teleporter message ID

Reply:
I think this is the receiving subnet and is the receipt from sending the `receiveCrossChainMessage` to destination chain.

Reply:
Yeah, this is the receiving subnet. `SendCrossChainMessage` and `ReceiveCrossChainMessage` happen to have the same fields, so the message unpacks without error.

Reply:
can we add a check to confirm with the current unpacking of `event.Destination.ChainID` to blockChainB? This should fail. 

Then we can update the unpacking to `receiveCrossCHainMessage` and check `originChainID`

Reply:
Yup, you're right. Fixed this now to retrieve the Teleporter message ID from the sending subnet logs as intended.

Reply:
I liked the check for unpacking the destination chain tx. Can we unpack the destination tx receipt log against the `receiveCrossChainMessage` and check it's `OriginChainID` and `MessageID`?